nodejs/week3/Mealsharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,202 @@
+const knex = require("../database");
+const express = require("express");
+const router = express.Router();
+//get method
+router.get("/meals", async (req, res) => {
+          try {
+                    const allMeals = await knex.from("meal").select("*");
+                    if (allMeals.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists any more"
+                              );
+                    }
+                    res.send(allMeals);
+          } catch (err) {
+                    throw err;
+          }
+});
+//get method using id
+router.get("/meals/:id", async (req, res) => {
+          try {
+                    const mealId = req.params.id;
```

Comment:
what happens if the `id` is not a number? How would you convey to the client that the params they sent are not valid?